File: templates/java-spark-monolith/src/main/java/SparkResources.java
import util.JsonTransformer;

import java.util.HashMap;
import java.util.Map;

import static spark.Spark.before;
import static spark.Spark.get;


public class SparkResources {

  public static void defineResources() {
    before((request, response) -> response.type("application/json"));

    get("/ping", (req, res) -> {
      Map<String, String> pong = new HashMap<>();
      pong.put("pong", "Hello, World!");
      res.status(200);
      return pong;
    }, new JsonTransformer());
  }

  public static void main(String[] args) throws Exception {
    SparkResources.defineResources();
  }
}

File: templates/kotlin-spark-monolith/src/main/kotlin/SparkResources.kt
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import spark.Filter
import spark.Request
import spark.Response
import spark.Spark.before
import spark.Spark.get
import java.util.*
import kotlin.collections.set


object SparkResources {

  fun defineResources() {

    val mapper = jacksonObjectMapper()

    before(Filter { _: Request, response: Response -> response.type("application/json") })

    get("/hello") { _: Request, res: Response ->
      val pong = HashMap<String, String>()
      pong["pong"] = "Hello, World!"
      res.status(200)
      mapper.writeValueAsString(pong)
    }

    get("/ping") { _: Request, res: Response ->
      val pong = HashMap<String, String>()
      pong["pong"] = "Ping!"
      res.status(200)
      mapper.writeValueAsString(pong)
    }
  }

  @Throws(Exception::class)
  @JvmStatic
  fun main(args: Array<String>) {
    SparkResources.defineResources()
  }
}